### PR TITLE
feat: add TVL calculation api

### PR DESCRIPTION
**Pull request by fubuloubu**
fixes: #113

### CODE REVIEW

File: contracts/BaseStrategy.sol [hunk 1]
```diff
@@ -149,6 +151,21 @@ abstract contract BaseStrategy {
...
     function name() external virtual pure returns (string memory);
 
+    /**
+     * @notice
+     *  The amount (priced in want) of the total assets managed by this strategy should not count
+     *  towards Yearn's TVL calculations.
+     * @dev
+     *  You can override this field to set it to a non-zero value if some of the assets of this
+     *  Strategy is somehow delegated inside another part of of Yearn's ecosystem e.g. another Vault.
+     *  Note that this value must be strictly less than or equal to the amonut provided by
```
Comment (dudesahn) [inline contracts/BaseStrategy.sol:163]: *amount

File: contracts/BaseStrategy.sol [hunk 2]
```diff
@@ -149,6 +153,21 @@ abstract contract BaseStrategy {
...
+    /**
+     * @notice
+     *  The amount (priced in want) of the total assets managed by this strategy should not count
+     *  towards Yearn's TVL calculations.
+     * @dev
+     *  You can override this field to set it to a non-zero value if some of the assets of this
+     *  Strategy is somehow delegated inside another part of of Yearn's ecosystem e.g. another Vault.
+     *  Note that this value must be strictly less than or equal to the amount provided by
+     *  `estimatedTotalAssets()` below, as the TVL calc will be total assets minus delegated assets.
+     * @return True if positions should be included in Yearn's Total Value Locked (TVL) calculation.
```
Comment (dmolina79) [inline contracts/BaseStrategy.sol:165]: Is this comment correct? Return value is uint

File: contracts/BaseStrategy.sol [hunk 3]
```diff
@@ -149,6 +153,21 @@ abstract contract BaseStrategy {
...
 
+    /**
+     * @notice
+     *  The amount (priced in want) of the total assets managed by this strategy should not count
+     *  towards Yearn's TVL calculations.
+     * @dev
+     *  You can override this field to set it to a non-zero value if some of the assets of this
+     *  Strategy is somehow delegated inside another part of of Yearn's ecosystem e.g. another Vault.
+     *  Note that this value must be strictly less than or equal to the amount provided by
+     *  `estimatedTotalAssets()` below, as the TVL calc will be total assets minus delegated assets.
```
Comment (dmolina79) [inline contracts/BaseStrategy.sol:164]: Not clear on the comment, is it correct to assume estimated total assets should be a sum of total assets in control of the strat + delegated assets? If this is the case we can make this non virtual and make the two sub method virtual? 

```suggestion
     *  EstimatedTotalAssets= 
TotalAssets() + delegatedAssets()
```

**Review by dmolina79 — APPROVED** [review]
Left two minor comments. Code looks good

File: contracts/BaseStrategy.sol (hunk 2 above)
Comment (dudesahn) [reply contracts/BaseStrategy.sol:165]: ah yeah I think that's probably leftover from when this was originally going to be boolean

Comment (fubuloubu) [reply contracts/BaseStrategy.sol:165]: crap, missed this

File: contracts/BaseStrategy.sol (hunk 3 above)
Comment (fubuloubu) [reply contracts/BaseStrategy.sol:164]: No, `delegatedAssets()` has a particular meaning for TVL calculations only. It means that amount is delegated to another Yearn component accounted for in TVL calculations, and as such should be subtracted from `estimatedTotalAssets()` here in order to conduct the calculation.

Comment (dmolina79) [reply contracts/BaseStrategy.sol:164]: Makes sense, thanks for explanation

**Pull request merged**